src/ocgis/conv/nc.py

```diff
@@ -97,8 +105,120 @@ def _raise_(msg, ocg_arugument=OutputFormat):
                 msg = 'Keyed function output may not be written to netCDF.'
                 _raise_(msg)
 
-    def _finalize_(self, ds):
-        pass
+        # Re-organize the collections following a discrete geometry model if aggregate is True
+        if ops.aggregate and not ops.geom:
+            msg = 'If aggregate is True than a geometry must be provided for netCDF output. '
+            _raise_(msg, OutputFormat)
+
+        if not ops.aggregate and not ops.agg_selection and ops.geom and len(ops.geom) > 1:
+            msg = 'Multiple geometries must either be unioned (agg_selection) ' \
+                  'or aggregated (aggregate).'
+            _raise_(msg, OutputFormat)
+
+
+    def _preformatting(self, i, coll):
+        """
+        Modify in place the collections so they can be saved as discrete
+        geometries along a new spatial dimension.
+        """
+        #TODO: UGID and GID show up in the output file, but they are equal. Remove one.
+
+        # Size of spatial dimension
+        if self.ops.aggregate is False:
```

Comment:
This is where it bails for the standard netCDF-CF-grid output, correct?

Reply:
Correct. The idea was to have _preformatting_ perform whatever operation is needed on the collection before sending it to the driver. For the CF-grid, there is nothing to do, so it just returns coll as is.

Reply:
Makes sense. Thanks.